### DISCUSSION TITLE
multiple open API spec files

### DISCUSSION THREAD

**Discussion by arunkumar413** [PaloAltoNetworks/docusaurus-openapi-docs#1060] [General]
Hi Team, 

I have multiple open API specification files and I would like to generate docs for them in a single docusaurus project. How can I achieve it? seems like `specPath` option in the config takes only the single file. 

Thanks,
Arun

**Comment by dmitriyrotaenko** [comment]
Hi, the `config` object of the `docusaurus-plugin-openapi-docs` plugin can accept multiple entry points for your APIs. See, the following example:

 ```
   {
        id: "api", // plugin id
        docsPluginId: "classic", // configured for preset-classic
        config: {
          first_api: {
            specPath: "path/to/first_api.yaml", // path to first api
            outputDir: "docs/first_api",
            sidebarOptions: {
              groupPathsBy: "tag",
            },
           second_api: {
              specPath: “path/to/second_api.yaml”, // path to second api
              …
           }
          }
        }
      }
```

**Reply by dmitriyrotaenko** [reply]
You can then generate docs by referring to `first_api`, `second_api`, or all apis that you defined in the `config` object.

CLI for docs generation is covered here https://docusaurus-openapi.tryingpan.dev/#generating-openapi-docs

**Reply by arunkumar413** [reply]
@dmitriyrotaenko Thanks a lot for the information.

**Comment by genosse-c** [comment]
Hi,

I'm also having issues with multiple APIs.

I have this plugin config:

<pre>plugins: [
    [
      "docusaurus-plugin-openapi-docs",
      {
				id: "api", // plugin id
				docsPluginId: "classic",
        config: {
          petstore: {
            specPath: "examples/petstore.yaml",
            outputDir: "docs/petstore",
            downloadUrl:
              "https://raw.githubusercontent.com/PaloAltoNetworks/docusaurus-template-openapi-docs/main/examples/petstore.yaml",
            sidebarOptions: {
              groupPathsBy: "tag",
              categoryLinkSource: "tag",
            },
          },
					other: {
            specPath: "examples/other.yaml",
            outputDir: "docs/other",
            downloadUrl:
              "https://raw.githubusercontent.com/PaloAltoNetworks/docusaurus-template-openapi-docs/main/examples/petstore.yaml",
            sidebarOptions: {
              groupPathsBy: "tag",
              categoryLinkSource: "tag",
            }          
          } satisfies OpenApiPlugin.Options,       
        } satisfies Plugin.PluginOptions,
      },  
    ],
  ],</pre>

My navbar config looks like this:

<pre>         {
            label: "Petstore API",
            position: "left",
            to: "/docs/category/petstore-api",
          },
          {
            label: "Other API",
            position: "left",
            to: "/docs/category/other-api",
          },      </pre>

However clicking on the "Other API" link results in a "page not found error". Any hint as to what I'm doing wrong?

Thanks a bunch!
